posixlib/src/main/resources/scala-native/sys/socket.c

```diff
@@ -67,8 +67,13 @@ _Static_assert(offsetof(struct scalanative_sockaddr, sa_data) ==
                    offsetof(struct sockaddr, sa_data),
                "offset mismatch: sockaddr sa_data");
 
+#if defined(__OpenBSD__)
+_Static_assert(sizeof(struct sockaddr_storage) == 256,
+               "unexpected size for sockaddr_storage");
+#else
```

Comment:
This one seems fishy, @LeeTibbert can you advise? 
AFAIR in the Scala Native in socket-stuff bindings we assume that it should have a size of 128bytes, otherwise we risk under-allocation and related with that undefined behaviour.

Reply:
good point, let me add more padding space into it.

Reply:
The .c code is doing its purpose, acting as a tripwire for porting.   Annoying as that is.
From a quick web search, I believe that `#ifdef OpenBSD` is needed. From the age of the
OpenBSD file I looked at, I am surprised that macOS does not need it also. 

The place that it is guarding is  posixlib `sys/socket.scala`.  The relevant code is:
```
 /* The declaration of sockaddr_storage should yield 128 bytes,                
   * with an overall alignment so that pointers have natural (64 )alignment.    
   */
  type sockaddr_storage = CStruct4[
    sa_family_t, // ss_family, // ss_family, sa_len is synthesized if needed    
    CUnsignedShort, // __opaquePadTo32                                          
    CUnsignedInt, // opaque, __opaquePadTo64                                    
    CArray[CUnsignedLongLong, _15] // __opaqueAlignStructure to 8 bytes.        
  ]

```
The last element is a fixed size array.   It is the `_15`  which is the issue.

I am sure that we do not want to unconditionally increase/double that for all operating systems.
Fine for private bootstrapping work. 

I think I/we used the "define the largest size" approach somewhere in `net` but I can not remember where.
There it was a few bytes, here we are talking 128, enough to cause concern.

Off the top of my head, I do not know of a way in Scala Native to use `LinkTimeMumble` or such to have define
a type differently depending upon the the operating system where compiled.

Wojciech do you know of a way for the Scala Native declaration to be one size on unix and another on OpenBSD?

Reply:
@LeeTibbert locally I allocate 256 bytes for all system and will add a comments. Semes that just a few tests is failed, and in a few hours, so, let make one PR which address general support of OpenBSD with everything.

I'll push the code into this PR in couple of hours. 16 failed tests remaining. BTW.

Reply:
I wish you energy, patience, & good fortune with the remaining failed tests. I've walked that path at
least twice.

Reply:
> Wojciech do you know of a way for the Scala Native declaration to be one size on unix and another on OpenBSD? 

We might use a trick used for zlib to define a different layout for Windows and Unix, but it makes usage harder then required. We might be able to fix it in the future when we'll land a better LLVM integration. Until that point doubling the Scala Native struct size is fine. After all we'd rather have ten/houndreads of socket allocatation, rather the millions so it should be fine for now.